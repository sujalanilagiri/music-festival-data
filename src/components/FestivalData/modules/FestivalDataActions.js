export const GET_FESTIVAL_DATA = "GET_FESTIVAL_DATA";

export const FESTIVAL_DATA = "FESTIVAL_DATA";

export const FESTIVAL_DATA_ERROR = "FESTIVAL_DATA_ERROR";
export function getFestivalData() {
  return {
    type: GET_FESTIVAL_DATA
  };
}

