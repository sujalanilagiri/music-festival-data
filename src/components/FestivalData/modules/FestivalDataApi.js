import axios from "axios";

const httpClient = axios.create();
httpClient.defaults.timeout = 150000;

export default class FestivalDataApi {
  static get() {
    
    return axios
      .get("/v1/festivals")
      .catch(function(error) {
        console.log('Api Call Failed',error)
      })
  }
  
}