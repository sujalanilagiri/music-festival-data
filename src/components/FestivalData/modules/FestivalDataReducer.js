import {
    GET_FESTIVAL_DATA,
    FESTIVAL_DATA,
    FESTIVAL_DATA_ERROR,
   } from "./FestivalDataActions";
  
  const FestivalDataReducer = (state = {}, action) => {
    switch (action.type) {
      case GET_FESTIVAL_DATA:
      case FESTIVAL_DATA:
       return {
          ...state,
           response:action.response
        };
        case FESTIVAL_DATA_ERROR:
        return {
           ...state,
            error:action.error
         };
      default:
        return state;
    }
  };
  
  export default FestivalDataReducer;