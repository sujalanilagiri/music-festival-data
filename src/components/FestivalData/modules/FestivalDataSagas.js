import { takeEvery, put, call } from "redux-saga/effects";
import FestivalDataApi from "./FestivalDataApi";


function* fetchData(action) {
  try {
    const response = yield call(FestivalDataApi.get);
    if (response.data) {
        yield put({ type: "FESTIVAL_DATA", response} );
    }
  } catch (error) {
    yield put({ type: "FESTIVAL_DATA_ERROR", error} );
  }
}

function* watchsaga() {
  yield takeEvery("GET_FESTIVAL_DATA", fetchData);
}

export const FestivalDataSagas = [watchsaga()];
