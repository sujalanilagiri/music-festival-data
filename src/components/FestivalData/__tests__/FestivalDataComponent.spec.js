import {configure} from 'enzyme';
import configureStore from "redux-mock-store";
import Adapter from 'enzyme-adapter-react-16';
import { getFestivalData } from "../modules/FestivalDataActions";
import FestivalDataReducer from "../modules/FestivalDataReducer";

let data =[
    {
      "name": "LOL-palooza",
      "bands": [
        {
          "name": "Jill Black",
          "recordLabel": "Fourth Woman Records"
        },
        {
          "name": "Werewolf Weekday",
          "recordLabel": "XS Recordings"
        },
        {
          "name": "Frank Jupiter",
          "recordLabel": "Pacific Records"
        },
        {
          "name": "Winter Primates",
          "recordLabel": ""
        }
      ]
    },
    {
      "name": "Small Night In",
      "bands": [
        {
          "name": "Squint-281",
          "recordLabel": "Outerscope"
        },
        {
          "name": "Yanke East",
          "recordLabel": "MEDIOCRE Music"
        },
        {
          "name": "Wild Antelope",
          "recordLabel": "Marner Sis. Recording"
        },
        {
          "name": "The Black Dashes",
          "recordLabel": "Fourth Woman Records"
        },
        {
          "name": "Green Mild Cold Capsicum",
          "recordLabel": "Marner Sis. Recording"
        }
      ]
    },
    {
      "name": "Trainerella",
      "bands": [
        {
          "name": "Manish Ditch",
          "recordLabel": "ACR"
        },
        {
          "name": "Adrian Venti",
          "recordLabel": "Monocracy Records"
        },
        {
          "name": "Wild Antelope",
          "recordLabel": "Still Bottom Records"
        },
        {
          "name": "YOUKRANE",
          "recordLabel": "Anti Records"
        }
      ]
    },
    {
      "name": "Twisted Tour",
      "bands": [
        {
          "name": "Auditones",
          "recordLabel": "Marner Sis. Recording"
        },
        {
          "name": "Summon",
          "recordLabel": "Outerscope"
        },
        {
          "name": "Squint-281"
        }
      ]
    },
    {
      "bands": [
        {
          "name": "Critter Girls",
          "recordLabel": "ACR"
        },
        {
          "name": "Propeller",
          "recordLabel": "Pacific Records"
        }
      ]
    }
  ]
configure({adapter:new Adapter()});

const users = [
    {
        "firstName": "Nicholas",
        "lastName" : "Thomas",
        "age": 42,
        "email": "nicholas.thomas@gmail.com"
    },
    {
        "firstName": "Elvin",
        "lastName" :"Martin",
        "age": 32,
        "email": "martin.elvin@gmail.com"
    },
    {
        "firstName": "Jass",
        "lastName" :"Base",
        "age": 22,
        "email": "jass.base@gmail.com"
    }
];
describe("ACTIONS: customerActions",()=>{
    const mockStore = configureStore();
    const store = mockStore();
    beforeEach(()=>{
        store.clearActions();
    });

    test("ACTION: GET_FESTIVAL_DATA",()=>{
        store.dispatch(getFestivalData());
        expect(store.getActions()).toMatchSnapshot();
    })
   
});

describe("REDUCER: FestivalDataReducer",()=>{
    test("GET_FESTIVAL_DATA",()=>{
        expect(FestivalDataReducer(undefined,getFestivalData(data))).toMatchSnapshot();
    })
})



