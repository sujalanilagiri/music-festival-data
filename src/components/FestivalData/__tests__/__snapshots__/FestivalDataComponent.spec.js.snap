// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ACTIONS: customerActions ACTION: GET_FESTIVAL_DATA 1`] = `
Array [
  Object {
    "type": "GET_FESTIVAL_DATA",
  },
]
`;

exports[`REDUCER: FestivalDataReducer GET_FESTIVAL_DATA 1`] = `
Object {
  "response": undefined,
}
`;
